src/promptflow/promptflow/executor/_process_manager.py

```diff
@@ -0,0 +1,327 @@
+import multiprocessing
+from multiprocessing import Queue
+from typing import List
+import queue
+import signal
+from functools import partial
+import psutil
+from promptflow._core.operation_context import OperationContext
+from promptflow._utils.logger_utils import LogContext, bulk_logger
+from promptflow.executor.flow_executor import FlowExecutor
+
+
+class AbstractProcessManager:
+    '''
+    AbstractProcessManager is a base class for managing processes.
+
+    :param input_queues: Queues for providing input data to the processes.
+    :type input_queues: List[multiprocessing.Queue]
+
+    :param output_queues: Queues for receiving execution results of the processes.
+    :type output_queues: List[multiprocessing.Queue]
+
+    :param process_info: Dictionary to store information about the processes.
```

Comment:
It would be more clear to use a dataclass instead of a dictionary for process info.